### PR TITLE
Add option to set a specific supabase URL on the server side

### PR DESCRIPTION
**Pull request by jfoucher**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

This change adds a new configuration option: `serverUrl`. When set, it overrides the `url` option on the serverside only when using SSR.

A  problem can occur when using nuxt-supabase with SSR in a docker-compose environment. The browser will connect to supabase on the exposed port, e.g. localhost:8000 whereas the server side, running in docker will have to connect to e.g. `supabase:8000`
This PR allows for setting a different supabase server URL on the browser side and on the server side.

## Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes (if not applicable, please state why)
I didn't see any tests in the source, nor a way to run any.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "@nuxtjs/supabase",
-  "version": "1.1.4",
+  "name": "jfoucher-nuxt-supabase",
```
Comment (Atinux) [inline package.json:2]: You should not update the `package.json`

**Comment by Atinux** [comment]
Could you please remove the update on the `package.json` and lock file?

Can you also update the documentation to explain about your new option? https://supabase.nuxtjs.org/get-started#options (https://github.com/nuxt-modules/supabase/edit/main/docs/content/2.get-started.md)

File: src/runtime/plugins/supabase.server.ts [hunk 2]
```diff
@@ -20,7 +20,7 @@ export default defineNuxtPlugin({
       }
     }, clientOptions)
 
-    const supabaseClient = createClient(url, key, options)
+    const supabaseClient = createClient(serverUrl ? serverUrl : url, key, options)
```
Comment (larbish) [inline src/runtime/plugins/supabase.server.ts:23]:
```suggestion
    const supabaseClient = createClient(serverUrl || url, key, options)
```

File: src/runtime/server/services/serverSupabaseClient.ts [hunk 3]
```diff
@@ -22,7 +22,7 @@ export const serverSupabaseClient = async <T>(event: H3Event): Promise<SupabaseC
         autoRefreshToken: false
       }
     }, clientOptions)
-    supabaseClient = createClient(url, key, options)
+    supabaseClient = createClient(serverUrl ? serverUrl : url, key, options)
```
Comment (larbish) [inline src/runtime/server/services/serverSupabaseClient.ts:25]:
```suggestion
    supabaseClient = createClient(serverUrl || url, key, options)
```

**Review by larbish — CHANGES_REQUESTED** [review]

File: src/runtime/plugins/supabase.server.ts (hunk 2 above)
Comment (jfoucher) [reply src/runtime/plugins/supabase.server.ts]: Fixed with the latest commit

**Comment by jfoucher** [comment]
Requested changes have been made in https://github.com/nuxt-modules/supabase/pull/319/commits/941ef0ee45bb24e61e71b8fd6742faaaea9ce1e4

**Pull request closed without merging**